powerdnsadmin/routes/index.py

```diff
@@ -170,6 +172,9 @@ def login():
         login_user(user, remember=False)
         session['authentication_type'] = 'OAuth'
         signin_history(user.username, 'Google OAuth', True)
+        if Setting().get('otp_first_login') and Setting().get('otp_field_enabled') and (user.otp_secret == "" or user.otp_secret is None):
```

Comment:
Can you put the multiple occurrences of these three lines into a new function that you call from multiple places and use a single block comment inside it?

You can simplify `(user.otp_secret == "" or user.otp_secret is None)` to `not user.otp_secret`.